packages/edit-navigation/README.md

```diff
@@ -59,13 +61,26 @@ This unlocks significant additional capabilities in the Navigation Editor. For e
 
 #### Technical Implementation details
 
-By default, `core/navigation-link` items are serialized and persisted as `nav_menu_item` posts. No serialized block HTML is stored for these standard link blocks.
+There are two types of item that can be added to a navigation:
```

Comment:
I wouldn't say there are only two things that can be added to a navigation because it is not true. I would say there are two kinds of navigations: classic, made of links, and block based made of blocks.

Because the editor is a block editor, we represent a classic link with a `core/navigation-link` A `core/navigation-link` block also represents a link when editing a block based menu.

This logic then applies below to the explanation: instead of the dichotomy "block based link" and "navigation link", use classic and block based contexts to explain what happens to the blocks in the editor, _normally_ when editing a classic menu and  _extra_ when editing a block based menu.

Reply:
> I would say there are two kinds of navigations: classic, made of links, and block based made of blocks.

Though even in a block based menu navigation link and navigation submenu blocks are saved in a different way to other blocks.

Reply:
Exactly. The idea is to make it clear that (currently) even if you used other types of blocks (ie: not `core/navigatoin-link`) the block data is stored using the existing menu system.

That might change but right now it is how it is.

Reply:
I really think the purpose of this section of the docs is to explain the difference between the types of _item_. Earlier in the docs I go into a lot of detail about the two "modes"

1. Classic
2. Block based.

I don't think we ought to repeat this again here.

Reply:
The more I thought about explaining the more I think we should skip over documenting something which is an experiment that we'll definitely deprecate (I mean the saving of blocks like HTML menu items)

Reply:
Right so basically we need to remove the entire `Technical Implementation details` section.

Reply:
Ok done. Noting that this PR was just intending to correct inconsistent information and errors so I'd rather not remove or rewrite any other large sections of docs within this PR if that's ok?